### DISCUSSION TITLE
Does Firefly III support end-to-end encryption in transit?

### DISCUSSION THREAD

**Discussion by sherif-fanous** [firefly-iii/firefly-iii#6097] [Firefly III]
I'm using HAProxy on pfSense as my reverse proxy to all my backend services.

HAProxy terminates the LetsEncrypt protected TLS connection. For all my existing backends I have HAProxy connect to them using a wildcard self-signed certificate for my internal/local domain name. So it basically looks like this

`Request <-- HTTPS (LetsEncrypt Certificate) --> HAProxy <-- HTTPS (Self Signed Certificate) --> Backend`

I have setup Firefly III as a backend and have configured the ACLs and Actions on the frontend to proxy the traffic to the Firefly III backend. So far so good and everything works as expected.

I would like (If possible) to set Firefly III to listen on HTTPS using my self-signed certificate similar to my other backends. I've looked through the documentation and I'm unable to figure out how to force Firefly III to do so.

The documentation states the following `Firefly III does not come with TLS enabled. This is up to you to setup. It does support it however.` and I'm not sure if I'm misinterpreting that Firefly III supports my use case or that Firefly III only supports direct HTTP but also supports working behind a reverse proxy with HTTPS enabled.

Thanks.

**Comment by JC5** [comment]
Firefly III supports it. It’s not really a matter of Firefly III supporting it, but rather the webserver.

As it is, the built-in webserver (Apache2) in the default Docker image does not have TLS enabled.  This is due to lack of demand and security posture. If you want to set this up, you’ll have to build the Docker container yourself and configure Apache as you need it.

**Comment by sherif-fanous** [comment]
I managed to set it up without having to build the Docker container myself.

Basically I override your Entry Point file to a new file that does some additional setup (Mainly enable the SSL module) and then invokes your Entry Point file.

Additionally I setup bind mounts to mount the needed files (New Entry Point file + Site Configuration with SSL + Cert & Key files) into the container.

**Reply by sherif-fanous** [reply]
That's the overridden Entry Point File

```
#!/bin/bash

echo "Now in entrypoint_ssl.sh for Firefly III"

echo "Enabling ssl module within apache2..."
a2enmod ssl

source /usr/local/bin/entrypoint.sh
```
